### PR TITLE
Adds continuous integration testing with travis 

### LINKED ISSUE #140: Add continuous integration testing
Set up with travis-ci so we can test each PR as they're created. We can [follow wc-admin's config.](https://github.com/woocommerce/wc-admin/blob/master/.travis.yml)

### LINKED ISSUE #153: Products & product info are not available to editors, block cannot show live preview
Users with role=editor or lower (author, contributor, etc) don't have the capabilities to edit/manage products or product categories, so the API doesn't let them list products. This causes the block preview to appear broken, never loading products or product categories (probably attributes too). The API response for these users is:

```
{
	"code":"woocommerce_rest_cannot_view",
	"message":"Sorry, you cannot list resources.",
	"data":{"status":403}
}
```

Should we…

- change the API to allow any logged in user to view products/product categories/attributes?
- Or should we only register the product block for admins & shop_managers (who already have permissions to see products etc)?

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #140 – this sets up travis to run JS and CSS linting, JS tests, and PHP tests, on the latest version of WP, across PHP 7.1 and 5.6. If this works out, I'll see about adding another job with WP 5.0.

I've added a short PHP test for the updated API endpoint, checking that the price ordering works. There's also a test checking that authors can't access product data, which is not the ideal behavior but can be used to help fix #153 

### How to test the changes in this Pull Request:

1. Whether the integration passes will be test enough, but eyes on the setup & test would be good.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+lts/*
```
Comment (ryelle) [inline .nvmrc]: This tells [`nvm`](https://github.com/creationix/nvm#nvmrc) to use the latest LTS release, currently 10.13.0 – I've also updated the package.json to reflect this as the node version.

File: package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "eslint-plugin-jsx-a11y": "6.1.2",
     "eslint-plugin-react": "7.11.1",
     "eslint-plugin-wordpress": "git://github.com/WordPress-Coding-Standards/eslint-plugin-wordpress.git#1774343f6226052a46b081e01db3fca8793cc9f1",
+    "har-validator": "^5.1.3",
```
Comment (ryelle) [inline package.json]: This fixes an issue downloading `har-validator@5.1.2`, because that package at that version doesn't exist anymore 🤷🏼‍♀️

Comment (timmyc) [reply package.json]: npm shenanigans!

File: tests/api/products-extra.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+ * @since 1.2.0
+ */
+class WC_Tests_API_Products_Controller extends WC_REST_Unit_Test_Case {
+
+	/**
+	 * Endpoints.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = '/wgbp/v3'; // @todo Fix endpoint typo?
```
Comment (timmyc) [inline tests/api/products-extra.php]: Perhaps we just make it a `/wc/v3` endpoint instead?

**Review by timmyc — APPROVED** [review]
Code is looking good to me, and CI appears to be happy!

**Pull request merged**